### PR TITLE
[IMPROVED] Protect against concurrent creation of streams and consumers.

### PR DESCRIPTION
**Pull request by derekcollison**
Also make sure we have exited monitoring routines when doing resets for both streams and consumers.

Signed-off-by: Derek Collison <derek@nats.io>

### CODE REVIEW

File: server/stream.go [hunk 1]
```diff
@@ -366,10 +366,21 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 		return nil, ApiErrors[JSStreamReplicasNotSupportedErr]
 	}
 
+	// Make sure we are ok when these are done in parallel.
+	var wg sync.WaitGroup
+	v, loaded := jsa.inflight.LoadOrStore(cfg.Name, &wg)
```
Comment (neilalexander) [inline server/stream.go:371]: Minor nit but `LoadOrStore(cfg.Name, &sync.WaitGroup{})` is clearer and doesn't pollute the scope with extra names.

**Review by neilalexander — APPROVED** [review]
LGTM!

**Review by Jarema — APPROVED** [review]
LGTM!

File: server/stream.go (hunk 1 above)
Comment (derekcollison) [reply server/stream.go:371]: Good call! Will update.

Comment (derekcollison) [reply server/stream.go]: Done.

**Review by wallyqs — APPROVED** [review]
LGTM

**Pull request merged**